cmd/cmd_util.go

```diff
@@ -0,0 +1,152 @@
+		if err != nil {
+			return fmt.Errorf("failed to update ReclaimPolicy for PV object %s: %v", pvc.Spec.VolumeName, err)
+		}
+		// Retrive old ceph volume name from PV
+		rbdImageName := kubernetes.GetFlexVolumeName(pv)
+		if rbdImageName == "" {
+			return fmt.Errorf("rbdImageName cannot be empty in Flex PV object: %v", pv)
+		}
+		logger.DefaultLog("flex rbdImageName name %v ", rbdImageName)
+
+		// Delete the PVC object
+		err = kubernetes.DeletePVC(client, &pvc) // nolint:gosec // skip gosec as pvc is accessed via it's reference.
+		if err != nil {
+			// TODO have an option to revert back if some Delete fails?
+			return fmt.Errorf("failed to Delete PVC object %s : %v", pvc.Name, err)
+		}
+		// Create new PVC with same name in destination storageclass
+		csiPVC := kubernetes.GenerateCSIPVC(destinationStorageClass, &pvc) // nolint:gosec // skip gosec as pvc is accessed via it's reference.
+		logger.DefaultLog("structure of the generated PVC in destination Storageclass \n %v ", csiPVC)
+
+		// Create new CSI PVC
+		csiPV, err := kubernetes.CreatePVC(client, csiPVC, 5)
+		if err != nil {
+			return fmt.Errorf("failed to Create CSI PVC object %s : %v", pvc.Name, err)
+		}
+		logger.DefaultLog("new PVC with same name created in CSI. Persistent Volume: %v ", csiPV)
+
+		// Extract new volume name from CSI PV
+		csiRBDImageName := kubernetes.GetCSIVolumeName(csiPV)
+		if csiRBDImageName == "" {
+			return fmt.Errorf("csiRBDImageName cannot be empty in PV object %v", csiPV)
+		}
+		logger.DefaultLog("csi new volume name: %v ", csiRBDImageName)
+		poolName := kubernetes.GetCSIPoolName(csiPV)
+		if poolName == "" {
+			return fmt.Errorf("poolName cannot be empty in PV object")
+		}
+		logger.DefaultLog("csi poolname: %v ", poolName)
+
+		// Create new connection
+		conn, err := createClusterConnection(client, csiPV)
+		if err != nil {
+			return fmt.Errorf("failed to get cluster config %v \n", err)
+		}
+		defer func() {
+			err = conn.Destroy()
+			if err != nil {
+				logger.ErrorLog("failed to destroy the connection: %v", err)
+			}
+		}()
+
+		// Delete the CSI volume in ceph cluster
+		err = conn.RemoveVolumeAdmin(poolName, csiRBDImageName)
+		if err != nil {
+			return fmt.Errorf("failed to delete the CSI volume in ceph cluster: %v", err)
+		}
+		logger.DefaultLog("successfully removed volume %s", csiRBDImageName)
+
+		// Rename old ceph volume to new CSI volume
+		err = conn.RenameVolume(csiRBDImageName, rbdImageName)
+		if err != nil {
+			return fmt.Errorf("failed to rename old ceph volume %s to new CSI volume %s: %v", rbdImageName, csiRBDImageName, err)
+		}
+		logger.DefaultLog("successfully renamed volume %s -> %s", csiRBDImageName, rbdImageName)
+
+		// Delete old PV object
+		err = kubernetes.DeletePV(client, pv)
+		if err != nil {
+			return fmt.Errorf("failed to delete persistent volume %s: %v", pv.Name, err)
+		}
+		logger.DefaultLog("deleted persistent volume %s", pv)
+	}
+
+	return err
```

Comment:
All errors are returned immediately inside the for loop, so no need to declare an err above, right? You can just return nil here.
```suggestion
	return nil
```